src/store/src/VectorStoreTrait.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Store;
+
+use Symfony\AI\Platform\Vector\Vector;
+use Symfony\AI\Store\Document\VectorDocument;
+use Symfony\AI\Store\Exception\InvalidArgumentException;
+
+/**
+ * @author Guillaume Loulier <personal@guillaumeloulier.fr>
+ */
+trait VectorStoreTrait
```

Comment:
does that have to be a trait? at first glance it looks like this could just be a standalone service class injected to both stores, or do I miss something here?

Reply:
i would argue the other way around - a separate class would be the easier choice as there is little to no cohesion but a functional dependency. and all the usual arguments would count - including testability.

let's go with a optional dependency like:
```php
    public function __construct(
        private readonly string $distanceCalculator = new DistanceCalculator(),
    ) {
    }
```

since the options are not part of that standalone contract, you could design the method quite explicit - even with an enum for the strategy

Reply:
Ah, I see, yes, didn't though about the optional dependency, this could open the possibility to extend it later with a custom interface, I like this approach 👍🏻

Reply:
Improved with a service / enum.

Reply:
Looks better now, thanks 👍